### PR TITLE
Week 3 nicole rivera

### PR DESCRIPTION
**Pull request by TsunamiCode5**
# Submitting Week 3 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ ✅] Coding Challenges
- [ ✅] Interview Questions
- [ ✅] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:
Again, I just wanted to apologize for the inconvenience of not getting my assignment done on time. I understand the work life balance is key to success in this career and I will not allow this to happen again. The one I think I still had most trouble with was #2 I had to crack open the notes to re-align myself but once I checked my thought process I got it quickly after. React is still my strongest suit. I would not like to know my progress report this week, just the normal feedback so I can bounce back into the groove of the week. Thank you for your awesome lectures!

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,114 @@
...
 
 // b) Create the function that makes the test pass.
 
+let nocchi = (number) => {
+        let fibArr = [1, 1]
+        for (let i=2; i < number; i++) {
+                fibArr.push(fibArr[i-2] + fibArr[i-1])
+        }
+        return fibArr
+}
```
Comment (elysemontano) [inline code-challenges.test.js:47]: Nice!  This looks great!

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,114 @@
...
+*/
 
 // b) Create the function that makes the test pass.
 
+// const leastToGreatest = (array) => {
+//         let filteredArr = array.filter(value => value % 2 !== 0 && typeof value === "number")
+// }
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// I had to look at my notes to give my path a direction
```
Comment (elysemontano) [inline code-challenges.test.js:79]: You are certainly encouraged to use your notes and the syllabus for these problems!  You got really far into this without your notes though, which is super impressive!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,114 @@
...
 // b) Create the function that makes the test pass.
+// const numbersToAdd1 = [2, 4, 45, 9]
+// const numbersToAdd2 = [0, 7, -8, 12]
+// const numbersToAdd3 = []
+// These were to check my console.logs
+
+const additionArray = (array) => {
+    let sum = 0
+    return array.map(value => sum += value)
+}
```
Comment (elysemontano) [inline code-challenges.test.js:125]: Excellent work!  Crushed this section Nicole!

File: interview-questions.md [hunk 4]
```diff
@@ -11,57 +11,83 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+  Researched answer:
 
+  What I explained was the use of 'this' in React, but even in React the main point is the 'this' is used in Class structures. React just uses a big class and so that was the easiest reference for me. The 'this' is the cellphones of Class houses. A way to call on specific information. The rooms of the house, are called objects.
```
Comment (elysemontano) [inline interview-questions.md:18]: Awesome analogy as always!

File: interview-questions.md [hunk 5]
```diff
@@ -11,57 +11,83 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
   Your answer:
 
-  Researched answer:
+  Yarn is used through your terminal to run, start, test, or create certain applications. Running React through your terminal with yarn start will allow you to use your local device to act as a network for you to see and interact with your application. It will refresh as soon as the file is saved, and giving that real time interaction allows you to modify and see your application as a user would once deployed. Yarn can also be used to test your code for cleaner code with less bugs. Yarn is also able to create a React App right through your terminal, using yarn create react-app. Files that are modified in a React application when you run yarn are those that are coded under the source folder or any of the given folders that are tied to React with "create react-app" with yarn as well.
 
+  Researched answer:
 
+  Yarn is the package manager. The files that are all nested within the package or included when you create or start React with Yarn, will be brought to your code editor. The files that are modified are the ones you have imported to your main app.js file.
```
Comment (elysemontano) [inline interview-questions.md:62]: Yarn packages up all of the node modules which contain a lot of prewritten behind-the-scenes code that allows our apps to run.  These are the files that will be modified when running yarn.

File: interview-questions.md [hunk 6]
```diff
@@ -11,57 +11,83 @@ Then, research the question to expand on your answer. Even if you feel you have
...
   Your answer:
 
+  An anonymous function is a function that hasn't been given a name but it is still a valid function. An example could be something like:
+  var hi = function () {
+    console.log('hello')
+    }
+
   Researched answer:
 
+  Most commonly an anonymous function is used as an argument to other functions. It is only accessible by the variable it stored in. An anonymous function can have multiply arguments. Giving it multiple arguments is a pro for developers to use anonymous functions in this way.
```
Comment (elysemontano) [inline interview-questions.md:75]: Anonymous functions are not stored to variables, as a result, they can only be called on only in the instance they are used.

**Review by elysemontano — APPROVED** [review]
Crushed this assessment Nicole!  I appreciate all the hard work you put into these assessments, and it certainly shows.  Your code is super efficient and clean, pseudocode is thorough and tests look fantastic.  Interview answers were outstanding, with great examples or analogies to support them.  Excellent work, keep it up!

File: interview-questions.md (hunk 4 above)
Comment (TsunamiCode5) [reply interview-questions.md:18]: This analogy actually helped like 5 people in class understand Classes in React better! Hence my uplifted mood last week.

**Pull request merged**